src/main/hook/patch/gfx.c

```diff
@@ -119,6 +189,25 @@ Window XCreateWindow(
       /* enables usage of nvidia cards where hardware colormaps are not available or supported */
       valuemask |= CWColormap;
     }
+    // Cache startup dimensions.
+    s_init_display_width = width;
+    s_init_display_height = height;
+    // If our target dimensions were zero, we will disable the flag to do any resizing.
+    if(s_target_display_height == 0){
+      s_target_display_height = height;
+    }
+    if(s_target_display_width == 0){
+      s_target_display_width = width;
+    }  
+    // We also need zoom factor
+    calculate_zoom_factors(s_init_display_width,s_init_display_height,s_target_display_width,s_target_display_height,&s_zoom_factor_x,&s_zoom_factor_y);  
+
+    // If we're not using the resizable window flag and our resolutions match, we won't do any adjustment.
+    if(s_resizable_window != 0 || s_target_display_width != width || s_target_display_height != height){
+      s_res_adjust_enabled = 1;
```

Comment:
That is a dead flag. It's not being used right now. I assume you wanted to use this instead of comparing the initial display width and height to the current width and height in `S3DResizeEx`.

I very much prefer having an explicit feature flag to switch this feature on/off instead of relying on implicit clues based on several parameters. It is more expressive and easier to understand and read in the code.

Since you already had this idea as well, do you mind switching to that?